### ISSUE TITLE
[TextInput][iPad] Numeric keyboard type not working

### ISSUE THREAD

**Issue by puradox** [facebook/react-native#17793]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

Environment:
  OS: macOS High Sierra 10.13.2
  Node: 9.3.0
  Yarn: 1.3.2
  npm: 5.5.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  react: 16.2.0 => 16.2.0
  react-native: 0.52.2 => 0.52.2

Target Platform: iOS (8.0)

### Steps to Reproduce

1. `react-native init NumbersPlease`
2. Open XCode project and change "General -> Devices" from "iPhone" to "iPad"
3. Modify `App.js` to render a `TextInput` with `keyboardType='numeric'`:
```js
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';
import {
  Platform,
  StyleSheet,
  Text,
  View,
  TextInput
} from 'react-native';

const instructions = Platform.select({
  ios: 'Press Cmd+R to reload,\n' +
    'Cmd+D or shake for dev menu',
  android: 'Double tap R on your keyboard to reload,\n' +
    'Shake or press menu button for dev menu',
});

export default class App extends Component<{}> {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit App.js
        </Text>
        <Text style={styles.instructions}>
          {instructions}
        </Text>
        <TextInput
          key='hi'
          autoFocus={true}
          keyboardType='numeric'
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});
```
4. `react-native run-ios --simulator "iPad Pro (12.9-inch) (2nd generation)"`

### Expected Behavior

A keyboard input automatically gaining focus with a **numeric** keyboard layout.

### Actual Behavior

A keyboard input automatically gaining focus with a **default** keyboard layout.

![simulator screen shot - ipad pro 12 9-inch 2nd generation - 2018-01-30 at 11 30 26](https://user-images.githubusercontent.com/941763/35587572-0c6a3a98-05b3-11e8-9fb7-0af82d7b46eb.png)

### Reproducible Demo

[puradox/NumbersPlease](https://github.com/puradox/NumbersPlease)

**Comment by jayeszee** [comment]
I think thats the numeric keyboard for ipads. On the phone it shows only numbers but I with an iPad its difficult to have an UI with just numbers. Can someone confirm this?

**Comment by gedeagas** [comment]
Yep, as @jayeszee mentioned this is the default behavior on iPad because apparently, it's "Hard" to have a UI with just numbers on iPad.

**Comment by salmanwaheed** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  mkdirenv@gmail.com
    Domain salmanwaheed.info has exceeded the max emails per hour (161/150 (107%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext3.iad.github.net ([192.30.252.194]:43643 helo=github-smtp2b-ext-cp1-prd.iad.github.net)
	by box1177.bluehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89_1)
	(envelope-from <noreply@github.com>)
	id 1epQUu-003Iys-0q
	for hello@salmanwaheed.info; Fri, 23 Feb 2018 20:27:40 -0700
Date: Fri, 23 Feb 2018 19:22:21 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1519442846;
	bh=FTJL8LNlb3O1bW0mYfEWBpMdf5oJT0vQZixvrpeqUBE=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=CdzYYgnX+KOhRkF71CLBQg7Oyy4T/tWmtM7f9PC51qDUlyeDCpQ4vEExCmj2tXInV
	 Mw5hLsF1ozyZybCgr+Tfp1Lk2R5S5N4Nq3vkpkr1Y1cgipl0XrxJzKfc6Pwc5Pw0RS
	 gJlU8XPa42Z80z9VhQrglQCjHVGwyaXHoP7cSpRs=

**Comment by aMarCruz** [comment]
@gedeagas , this bug is present in Android for RN 0.53.3

**Comment by nimamyscreen** [comment]
I'm using react native 0.55.2 and I still have this problem in IPad

**Comment by saggiyogesh** [comment]
I'm using 0.56.0. Same problem.
Think it's an iOS related. 
https://stackoverflow.com/questions/27409483/number-keyboard-in-ipad
